packages/portalnetwork/src/wire/utp/Protocol/utp_protocol.ts

```diff
@@ -52,68 +58,122 @@ export class UtpProtocol {
     }
   }
 
+  async initiateUtpTest(remoteAddr: string, connectionId: number, networkId: SubNetworkIds) {
+    this.log(`Requesting uTP stream connection with ${remoteAddr}...`)
+    const socket = new _UTPSocket(this, remoteAddr, 'writing', networkId, 0)
+    socket.content = testArray
+    this.sockets[remoteAddr + connectionId] = socket
+    await this.sockets[remoteAddr + connectionId].sendSynPacket(connectionId)
+  }
+
   /**
-   * Creates Write Socket with content from OFFER/ACCEPT
+   * Creates Write Sockets for each piece of content from OFFER/ACCEPT
    * @param remoteAddr
-   * @param connectionId
+   * @param connectionId uTP ID set by accepting node
    * @param contentKeys payload from ACCEPT message
+   * @param contentTypes array of content types accepted
+   * @param networkId
    */
 
-  async initiateUtpFromAccept(
+  async handleAccept(
     remoteAddr: string,
     connectionId: number,
     contentKeys: Uint8Array[],
+    contentTypes: number[],
     networkId: SubNetworkIds
   ) {
-    const socket = new _UTPSocket(this, remoteAddr, 'writing', networkId)
-    const value = await this.portal.db.get(serializedContentKeyToContentId(contentKeys[0]))
-    socket.content = fromHexString(value)
-    this.sockets[remoteAddr + connectionId] = socket
-    await this.sockets[remoteAddr + connectionId].sendSynPacket(connectionId)
+    const socketKey = this.getSocketKey(remoteAddr, connectionId)
+    contentKeys.forEach(async (content, idx) => {
```

Comment:
Does this `forEach` imply we're creating multiple sockets (one per `contentType`) but that share the same `socketKey`?  It's not 100% clear to me why we're passing the `desiredContentTypes` as an array here and if this is attempting to create multiple sockets or not (since it doesn't look like it will since).  If we're using only a single connectionId, our current use case for uTP in the `OFFER`/`ACCEPT` flow implies we'll only send a single piece of content over the uTP socket (corresponding to a single `contentKey`).  And we can derive the `contentType` for the socket from the `contentKey`, right?

Reply:
Yes to the 'single piece of content' per socket thing.  For each piece of accepted content it will make a new socket.  And yes, sharing the socketKey is an unsolved problem that arises from that.  I see what you mean about the socket deriving the contentType for itself.  Yes, that would be simpler than saving the array of types.